### ISSUE TITLE
Callback not called when still animating

### ISSUE THREAD

**Issue by hellsan631** [t4t5/sweetalert#528]
I integrated sweetalerts into my angular logger framework, and i love promises. I'm trying to use a webkit-filter to blur when a modal is open, and came across this bug.

``` js
function errorDialog(title, text) {
  var deferred = $q.defer();

  $rootScope.modalOpen = true;

  swal({
    title: title,
    text: text,
    type: 'error'
  }, resolve);

  deferred.promise
    .then(function(){
      $rootScope.modalOpen = false;
     });

  return deferred.promise;

  function resolve() {
    console.log('res');
    deferred.resolve();
  }
}
```

When ever i close the modal too early, before the little error cross icon animation, the callback never fires. this most likely is because the callback function is called on a button click event which is bound after a non-async animation call, and not on the dialog teardown.
# 

As a small node, I handle this now by binding a setInterval that checks if the .sweetalert element has the class hideSweetAlert, which then fires the callback, instead of relying on the callback of swal.

**Comment by rafael-renan-pacheco** [comment]
Same problem here too, easily reproducible.

I have a "Delete" button which fires a confirm dialog using swal. The swal confirm button takes the same place as the "Delete" button behind it, making it easy to click on it before swal animation is completed.

When this occur, the confirm callback is not fired. In my case, swal own animation is blocking the callback when it has not yet finished.

**Comment by hellsan631** [comment]
Also, the callback never fires when you hit the esc key.

To remedy both situations, here is some rancid code.

```js
function attachKeyDown(fn) {
  var check = setInterval(function() {
    if ($('.sweet-alert').hasClass('hideSweetAlert')) {
      callback();
    }
  }, 100);
  
  document.onkeydown = function(evt) {
    evt = evt || window.event;
    if (evt.keyCode == 27) {
      callback();
    }
  };

  function callback() {
    document.onkeydown = null;
    clearInterval(check);
    fn();
  }
}
```

It calls the callback function on the esc key down and also if the class 'hideSweetAlert' is on the sweet alert container div. Its not optimal, but it works.

**Comment by alvarotrigo** [comment]
+1

**Comment by t4t5** [comment]
Fixed in [2.0](https://sweetalert.js.org) (now with promises!)
